source/routing/raptor_api.cpp

```diff
@@ -1469,15 +1460,39 @@ void modify_backtracking_journeys(RAPTOR::Journeys& journeys,
                                   const bool clockwise) {
     for (auto& journey : journeys) {
         if (clockwise) {
-            auto last_section = journey.sections.rbegin();
+            const auto& last_section = journey.sections.rbegin();
             const auto& last_stop_area_uri = last_section->get_out_st->stop_point->stop_area->uri;
             const auto* last_vj = last_section->get_out_st->vehicle_journey;
-            shorten_journey(journey, last_stop_area_uri, last_vj, departures, destinations, clockwise);
+
+            size_t section_idx = 0;
+            // browse sections and modify any section that allows shortening
+            for (; section_idx < journey.sections.size(); section_idx++) {
```

Comment:
For clockwise, you should iterate sections backward and stops at the first section that can be shorten.